data_safe_haven/utility/annotated_types.py

```diff
@@ -22,3 +24,9 @@
 Guid = Annotated[str, AfterValidator(validate_aad_guid)]
 IpAddress = Annotated[str, AfterValidator(validate_ip_address)]
 TimeZone = Annotated[str, AfterValidator(validate_timezone)]
+TH = TypeVar("TH", bound=Hashable)
+# type UniqueList[TH] = Annotated[list[TH], AfterValidator(validate_unique_list)]
+# mypy doesn't support PEP695 type statements
+UniqueList: TypeAlias = Annotated[  # noqa:UP040
+    list[TH], AfterValidator(validate_unique_list)
+]
```

Comment:
Why do we need this instead of `UniqueList: TypeAlias = Annotated[list[Hashable], AfterValidator(validate_unique_list)]`?

Reply:
I think that gave me a typing error like "expected `list[Hashable]` but got `list[int]`". So I think that type annotation would be "A list with items of the type class: Hashable" rather than "A list of objects where `isinstance(x, Hashable) == True`".

Maybe I did something wrong.

Reply:
OK, I guess the only other solution would be to put `x: list[Hashable] = [1, 2, 3]` everywhere which is *much* worse. I'm not totally sure why the current solution works, but I'm glad it does!

Reply:
@jemrobinson Happy to accept what is here or can you see a way around that?

Reply:
Definitely happy to accept this. See below for my remaining question about whether we should just be using `set`.